Lib/fontbakery/commands/check_spec.py

```diff
@@ -39,165 +39,27 @@
-                      'In json output: structure the document by ITERATED_ARG.\n'
-                      'One of: {}'.format(', '.join(gather_by_choices))
-                      )
-
-  def parse_order(arg):
-    order = filter(len, [n.strip() for n in arg.split(',')])
-    return order or None
-  argument_parser.add_argument('-o','--order', default=None, type=parse_order,
-                      help='Comma separated list of order arguments.\n'
-                      'The execution order is determined by the order of the test\n'
-                      'definitions and by the order of the iterable arguments.\n'
-                      'A section defines its own order. `--order` can be used to\n'
-                      'override the order of *all* sections.\n'
-                      'Despite the ITERATED_ARGS there are two special\n'
-                      'values available:\n'
-                      '"*iterargs" -- all remainig ITERATED_ARGS\n'
-                      '"*test"     -- order by test\n'
-                      'ITERATED_ARGS: {}\n'
-                      'A sections default is equivalent to: "*iterargs, *test".\n'
-                      'A common use case is `-o "*test"` when testing the whole \n'
-                      'collection against a selection of tests picked with `--checkid`.'
-                      ''.format(', '.join(iterargs))
-                      )
-  return argument_parser
-
-class ArgumentParserError(Exception): pass
-
-class ThrowingArgumentParser(argparse.ArgumentParser):
-    def error(self, message):
-        raise ArgumentParserError(message)
-
-def get_spec():
-  """ Prefetch the specification module, to fill some holes in the help text."""
-  argument_parser = ThrowingArgumentParser(add_help=False)
-  argument_parser.add_argument('specification')
-  try:
-    args, unknown = argument_parser.parse_known_args()
-  except ArgumentParserError as e:
-    # silently fails, the main parser will show usage string.
-    return Spec()
-
-  from importlib import import_module
-  # Fails with an appropriate ImportError.
-  imported = import_module(args.specification, package=None)
-  # Fails with an attribute error if specification is undefined.
-  return imported.specification
-
-def runner_factory(specification, fonts, explicit_tests=None, custom_order=None
+def runner_factory(specification, label, items, explicit_tests=None, custom_order=None
                                                         , values=None):
   """ Convenience TestRunner factory. """
-  values_ = dict(fonts=fonts)
+  values_ = {label: items}
   if values is not None:
     values_.update(values)
   return TestRunner( specification, values_
                    , explicit_tests=explicit_tests
                    , custom_order=custom_order
                    )
 
-def main(specification=None, values=None):
-  # this won't be used in check-googlefonts
-  add_spec_arg = False
-  if specification is None:
```

Comment:
This is a cool feature! And it's the whole point of the check_spec module.

Reply:
That means we'll work to get it back.

Reply:
The idea was that we can make a command that sets up all but the actual specification to run. So that a specification implementor wouldn't have to set up an own bin command, but calll this generic one, with instructions from where to import a spec. 

Your problem here starts with you writing a spec that is not interested in the `fonts` argument. But you end up changing an interface and breaking its intend and its dependants.


To do what you want: 

1) roll back the changes in here. 
2) let the specification define it's custom arguments itself
3) keep the more generic arguments in here, like the output control (Loglevel). If we need this to be more customized, it's probably better to create a  new command from scratch.